frontend/pages/queries/ManageQueriesPage/ManageQueriesPage.tsx

```diff
@@ -162,7 +168,10 @@ const ManageQueriesPage = ({
     }
   );
 
-  const enhancedQueries = queriesResponse?.queries.map(enhanceQuery);
+  // Enhance the queries from the response when they are changed.
+  const enhancedQueries = useMemo(() => {
+    return queriesResponse?.queries.map(enhanceQuery) || [];
+  }, [queriesResponse]);
```

Comment:
It seems like the better way to do this would be to create this map in a `select` function in `useQuery`, but this code seems to have evolved away from that pattern and it would be much higher touch to replace it.